src/Approximations/overapproximate.jl

```diff
@@ -166,3 +166,27 @@ function overapproximate(S::LazySet{N}, ::Type{Interval}) where {N<:Real}
     hi = σ([one(N)], S)[1]
     return Interval(lo, hi)
 end
+
+# overapproximate an intersection with a polytope in H-representation given a
+# set of template directions.
+function overapproximate(cap::Intersection{N, <:LazySet, S},
+                         dir::AbstractDirections{N};
+                         kwargs...) where {N<:Real, S<:AbstractPolytope{N}}
+
+    X = cap.X    # compact set
+    P = cap.Y    # polytope
+
+    clist = constraints_list(P)
+    m = length(clist)
+    Q = HPolytope{N}()
+    ρ_X_Hi = Vector{N}(undef, m)
+
+    for di in dir
+        for (i, Hi) in enumerate(clist)
+            ρ_X_Hi[i] = ρ(di, X∩Hi, kwargs...)
+        end
+        i_min = indmin(ρ_X_Hi)
```

Comment:
Since you are only interested in the minimum, it suffices to store *one* index and *one* support value. Just add the minimum check inside the loop.